community/blog/2023-02 Integrating Arcus Security in a non-invasive manner/README.md

```diff
@@ -0,0 +1,134 @@
+
+```csharp
+using Arcus.Security.Core;
+using Azure.Storage;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+
+var host = new HostBuilder()
+    .ConfigureFunctionsWorkerDefaults()
+    .ConfigureAppConfiguration(builder =>
+    {
+        builder.AddEnvironmentVariables();
+    })
+    .ConfigureSecretStore((config, stores) =>
+    {
+        stores.AddEnvironmentVariables();
+    })
+    .ConfigureServices((context, services) =>
+    {
+        services.AddAzureClients(clients =>
+        {
+            string serviceUrl = context.Configuration["AZURE_BLOB_URL"];
+            string accountName = context.Configuration["AZURE_BLOB_ACCOUNTNAME"];
+
+            clients.AddClient<BlobServiceClient, BlobClientOptions>((options, serviceProvider) =>
+            {
+                var secretProvider = serviceProvider.GetRequiredService<ISecretProvider>();
+                string accountKey = secretProvider.GetRawSecret("AZURE_BLOB_ACCOUNTKEY");
+
+                return new BlobServiceClient(new Uri(serviceUrl), new StorageSharedKeyCredential(accountName, accountKey));
+            });
+        });
+    })
+    .Build();
+
+host.Run();
+```
+
+Notice that the Azure Functions HTTP trigger does not have to change, it is only the retrieval of secrets that has to be adapted. At this point, one may wonder why we initialize both the application configuration and the secret store with the same source (environment variables). And that is a good thing! This means that we start to think of our input as different kinds of types. Both the Arcus environment secret provider and the Microsoft environment variable configuration source can retrieve only a portion of the environment variables by specifying a prefix. This could be a great way to already make the distinction right from the start between application configuration and sensitive secrets.
+
+## Conclusion
+This post showed a way to integrate the Arcus secret store into an existing application without being evasive of the actual application code. It is by design that the example uses Azure Blob storage, as this is an Azure technology we do not necessarily work upon in Arcus (like we do with Service Bus or Event Hubs). It is proof that Arcus can be used in any number of scenarios and should not be limited by the technologies described in our feature documentation.
+
+This example shows how we can separate application configuration from sensitive secrets. This is an important step in making our application more secure and safer to use during future development. Treating secrets as secrets is a mentality shift that will bring great value as it will force you to think about your inputs differently. Not all inputs are public data and should be treated accordingly. Furthermore, we should place the secrets in a secure place (like Azure Key vault) instead of storing secrets and application configuration in the same place.
```

Comment:
Maybe also mention that now the ISecretProvider is in place, it is quite easy to do so, as we can simply register the KeyVault SecretProvider instead, and don't need to change anything else to the code.

Reply:
Mentioned this and added link to Azure Key vault secret provider. 👍